app/server/jsondb/src/main/java/io/syndesis/server/jsondb/dao/Migrator.java

```diff
@@ -22,4 +22,5 @@
  */
 public interface Migrator {
     void migrate(JsonDB jsondb, int schema);
+    void migrate(JsonDB jsondb, int schema, String migrationDir);
```

Comment:
Perhaps it would be better to have just the one method on the Migrator, something like:

`void migrate(JsonDB jsondb, int schema, String... baseResources)`

And the `resources` parameter would be in Spring [resource abstraction syntax](https://docs.spring.io/spring/docs/5.1.0.BUILD-SNAPSHOT/spring-framework-reference/core.html#resources-resource-strings).

So we could say here `classpath*:/migrations/` or `/path/to/dir/`.